compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -1021,15 +1021,19 @@ private STNode parseVarDeclWithFunctionType(STNode functionKeyword, STNode funcS
         STNode visibilityQualifier = STNodeFactory.createEmptyNode();
         List<STNode> validatedQualifierList = new ArrayList<>();
         // qualifiers are only allowed in the following cases for func type desc.
-        // isolated and transactional qualifiers allowed.
-        // public or private qualifier allowed in object field.
+        // isolated and transactional qualifiers are allowed.
+        // public, private and remote qualifiers are allowed in object field.
         for (int position = 0; position < qualifierList.size(); position++) {
             STNode qualifier = qualifierList.get(position);
             if (qualifier.kind == SyntaxKind.ISOLATED_KEYWORD || qualifier.kind == SyntaxKind.TRANSACTIONAL_KEYWORD) {
                 validatedQualifierList.add(qualifier);
-            } else if (isObjectMember && isVisibilityQualifier(qualifier)) {
-                // public or private qualifier allowed in object field.
-                visibilityQualifier = qualifier;
+            } else if (isObjectMember) {
+                if (isVisibilityQualifier(qualifier)) {
+                    // public or private qualifier allowed in object field.
+                    visibilityQualifier = qualifier;
+                } else if (qualifier.kind == SyntaxKind.REMOTE_KEYWORD) {
+                    validatedQualifierList.add(qualifier);
+                }
```

Comment:
do we need a `else` block? i.e: when the qualifier is neither a visibility qualifier nor REMOTE_KEYWORD

Reply:
oops!